File: matlab/PCAmat2feat.m
function PCAmat2feat(curdir,codedir,MatListFile,FeatListFile,reduceDim,bUseSavedPCAfeatMatrix,LogfileExt)

path(path,codedir);

MaxMatList=2000;

%bUseSavedPCAfeatMatrix=str2num(bUseSavedPCAFlag);
%###PCAfeat###
if bUseSavedPCAfeatMatrix==1
	disp('WARNING: PCAfeatmodel loaded from file PCAfeatmodel.mat !!! (see PCAmat2feat.pl)');
	load([curdir,'/PCAfeatmodel.mat'],'PCAfeatmodel');
else
%###PCAfeat###
	disp('CALCULATING PCAfeatmodel from MaxMatList random entries in MatListFile');
	COVA=GetScatterMatrix(MatListFile,curdir,MaxMatList);
	PCAfeatmodel=pca_cov(COVA,reduceDim);
	save([curdir,'/PCAfeatmodel.mat'],'PCAfeatmodel');
	disp('PCAfeatmodel saved in file PCAfeatmodel.mat saved in current dir');
end

PCAmatrixW=PCAfeatmodel.W;
ProjectData(PCAmatrixW,MatListFile,FeatListFile,curdir)

fid_LogfileExt=fopen(LogfileExt,'w');
fprintf(fid_LogfileExt,'%s\n',[LogfileExt,' done!']);
fclose(fid_LogfileExt);

File: perl/PCAmat2feat.pl
#!/usr/bin/perl
use strict;

#echo PCAmat2feat.pl currdir matlabdir matlist featlist reduceDim
#mat dir $1/feature_mat/

my $curdir		= $ARGV[0];
my $codedir		= $ARGV[1];
my $MatListFile	= $ARGV[2];
my $FeatListFile= $ARGV[3];
my $reduceDim	= $ARGV[4];
my $bUseSavedPCAfeatMatrix = $ARGV[5];

my $Logfile = "$curdir/matlab.PCAmat2feat.out";
my $LogfileExt = "$curdir/matlab.PCAmat2feat.out.ext";

open(FILE,'>PCAmat2feat_run.m') || die @_;
print FILE "path(path,\'$codedir\')\n";
print FILE "PCAmat2feat(\'$curdir\',\'$codedir\',\'$MatListFile\',\'$FeatListFile\',$reduceDim,$bUseSavedPCAfeatMatrix,\'$LogfileExt\')\n";
print FILE "exit\n";
close(FILE);

system("matlab -nosplash -nodesktop -r PCAmat2feat_run -logfile $Logfile");

